### DISCUSSION TITLE
Using non-string tags as strings in path name

### DISCUSSION THREAD

**Discussion by changwoo** [dcm4che/dcm4che#1240] [Q&A]
**Describe the bug**
Please add a feature to use some non-string tags as strings in AttributesFormat.

**To Reproduce**
1. Run storescp with --filepath and private tags in the path format.
2. Send dicom with the encoded private tags with storescu.
3. private tags are not strings, so they are not properly decoded.

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: All
 - Browser N/A
 - Version 5.16.0

**Additional context**
In my institution, there are some researchers who want to use private tags for file pathnames. For example:
```
  storescp.bat .... --filepath "{00091080}/{00080018}.dcm"
```
, where the private tag (0009,1080) usually contains a Korean encoded string. Then "{00091080}" is not formatted to a properly decoded string. I know it is because that this tag is not for a string and SpecificCharacterSet encoding is not applicable. But some devices allow Korean string input in private tags like this, so it would be useful if we could use some tags as encoded strings.

How about adding a type specifier to assume a tag as a string?  For example, like this:
```
  storescp.bat ....  --filepath "{00091080,string}/{00080018}.dcm"
```

**Comment by vrindanayak** [comment]
@changwoo : Can you share sample testdata (DICOM file)?

**Comment by changwoo** [comment]
@vrindanayak unfortunately my current samples are not legally distributable. Instead I'm going to make one. Do you have an advise on how to make a sample dicom with desired tags quickly? I'm still a newbie on dicom tools.

**Comment by vrindanayak** [comment]
@changwoo : Alternatively to retry the issue you reported, you may create a dictionary specific to your requirements (this dictionary shall extend the dcm4che library's ElementDictionary), which maps your private tags to specific string value representations, and add it in the class path.

For reference, see [Siemens private tags dictionary](https://github.com/dcm4che/dcm4che/tree/master/dcm4che-core/src/test/java/org/dcm4che3/dict/siemens) registered as a [service](https://github.com/dcm4che/dcm4che/blob/master/dcm4che-core/src/test/resources/META-INF/services/org.dcm4che3.data.ElementDictionary)

**Comment by gunterze** [comment]
You may convert non-ascii string values to `%XX%YY%ZZ` with `XX`, `YY`,  `ZZ` the hex value of the first, second, third byte of the UTF-8 code of the character (e.g.. `洪` -> `%E6%B4%AA`) by `{00091080,urlencoded}`.
